src/linalg/impl_linalg.rs

```diff
@@ -78,7 +78,7 @@ where
         let mut sum = A::zero();
         for i in 0..self.len() {
             unsafe {
-                sum = sum.clone() + self.uget(i).clone() * rhs.uget(i).clone();
+                sum = sum + *self.uget(i) * *rhs.uget(i);
```

Comment:
Unsafe so needs a review

Reply:
Looks good to me. (It's just changing `.clone()` calls to copies.)